NullPointerException using Print to Host for Pixel prints

When using Print To Host, throws a `NullPointerException`

* OS: Ubuntu 22.04
* QZ Tray 2.1.6+3, 2.2.1, 2.2.2-SNAPSHOT
* Java 11.0.+8

```
[DEBUG] 2022-12-13T09:37:03,845 @ qz.ws.PrintSocketClient:-1
	Message: {"call":"print","params":{"printer":{"host":"192.168.1.254","port":"9100"},"options":{"bounds":null,"colorType":"color","copies":1,"density":0,"duplex":false,"fallbackDensity":null,"interpolation":"bicubic","jobName":null,"legacy":false,"margins":0,"orientation":null,"paperThickness":null,"printerTray":null,"rasterize":false,"rotation":0,"scaleContent":true,"size":null,"units":"in","encoding":null,"spool":{},"altPrinting":false},"data":[{"type":"pixel","format":"pdf","flavor":"file","data":"https://demo.qz.io/assets/pdf_sample.pdf","options":{"pageWidth":"","pageHeight":"","pageRanges":"","ignoreTransparency":false}}]},"signature":"","timestamp":1670942223843,"uid":"wdnnb8","position":{"x":757,"y":463.5},"signAlgorithm":"SHA512"}
[WARN] 2022-12-13T09:37:03,848 @ qz.ws.PrintSocketClient:-1
	Bad signature on request
[DEBUG] 2022-12-13T09:37:03,916 @ qz.utils.GtkUtilities:-1
	Finding preferred Gtk version...
[DEBUG] 2022-12-13T09:37:03,921 @ qz.utils.GtkUtilities:-1
	Found Gtk2
[DEBUG] 2022-12-13T09:37:03,921 @ qz.utils.GtkUtilities:-1
	Initialized Gtk
[DEBUG] 2022-12-13T09:37:03,922 @ qz.utils.GtkUtilities:-1
	Gtk 2.10+ detected, calling "gdk_screen_get_resolution"
[DEBUG] 2022-12-13T09:37:03,922 @ qz.common.TrayManager:-1
	Calculated dialog centered at: java.awt.Point[x=577,y=374]
[INFO] 2022-12-13T09:37:05,097 @ qz.common.TrayManager:-1
	Allowed localhost to print to 192.168.1.254
[TRACE] 2022-12-13T09:37:05,097 @ qz.common.TrayManager:-1
	Not idle, stopping any actions that haven't ran yet
[DEBUG] 2022-12-13T09:37:05,120 @ qz.utils.PrintingUtilities:-1
	Allowing 1 simultaneous processors based on memory available (536 MB)
[TRACE] 2022-12-13T09:37:05,120 @ qz.utils.PrintingUtilities:-1
	Waiting for processor, 0/1 already in use
[DEBUG] 2022-12-13T09:37:05,137 @ qz.utils.PrintingUtilities:-1
	Using qz.printer.action.PrintPDF to print
[WARN] 2022-12-13T09:37:05,145 @ qz.printer.PrintOptions:-1
	Cannot read null as a double for fallbackDensity, using default
[DEBUG] 2022-12-13T09:37:06,159 @ qz.printer.action.PrintPDF:-1
	Parsed 1 files for printing
[ERROR] 2022-12-13T09:37:06,185 @ qz.utils.PrintingUtilities:-1
	Failed to print
java.lang.NullPointerException: null
	at qz.printer.PrintOutput.getPrintService(Unknown Source) ~[qz-tray.jar:?]
	at qz.printer.action.PrintPDF.print(Unknown Source) ~[qz-tray.jar:?]
	at qz.utils.PrintingUtilities.processPrintRequest(Unknown Source) ~[qz-tray.jar:?]
	at qz.ws.PrintSocketClient.processMessage(Unknown Source) ~[qz-tray.jar:?]
	at qz.ws.PrintSocketClient.onMessage(Unknown Source) ~[qz-tray.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.eclipse.jetty.websocket.common.events.annotated.CallableMethod.call(CallableMethod.java:70) ~[qz-tray.jar:?]
	at org.eclipse.jetty.websocket.common.events.annotated.OptionalSessionCallableMethod.call(OptionalSessionCallableMethod.java:68) ~[qz-tray.jar:?]
	at org.eclipse.jetty.websocket.common.events.JettyAnnotatedEventDriver.lambda$onTextFrame$1(JettyAnnotatedEventDriver.java:270) ~[qz-tray.jar:?]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883) ~[qz-tray.jar:?]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034) ~[qz-tray.jar:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
[TRACE] 2022-12-13T09:37:06,201 @ qz.utils.PrintingUtilities:-1
	Returning processor back to pool
```

--- a/src/qz/utils/PrintingUtilities.java
+++ b/src/qz/utils/PrintingUtilities.java
@@ -17,7 +17,6 @@ import qz.ws.PrintSocketClient;
 
 import java.awt.print.PrinterAbortException;
 import java.io.IOException;
-import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Locale;
@@ -90,29 +89,21 @@ public class PrintingUtilities {
         }
     }
 
-
-    public static Format getPrintFormat(JSONArray printData) throws JSONException {
-        convertVersion(printData);
-
-        //grab first data object to determine type for entire set
-        JSONObject data = printData.optJSONObject(0);
-
-        //Check for RAW type to coerce COMMAND format handling
-        Type type;
+    public static Type getPrintType(JSONObject data) {
         if (data == null) {
-            type = Type.RAW;
+            return Type.RAW;
         } else {
-            type = Type.valueOf(data.optString("type", "RAW").toUpperCase(Locale.ENGLISH));
+            return Type.valueOf(data.optString("type", "RAW").toUpperCase(Locale.ENGLISH));
         }
+    }
 
-        Format format;
+    public static Format getPrintFormat(Type type, JSONObject data) {
+        //Check for RAW type to coerce COMMAND format handling
         if (type == Type.RAW) {
-            format = Format.COMMAND;
+            return Format.COMMAND;
         } else {
-            format = Format.valueOf(data.optString("format", "COMMAND").toUpperCase(Locale.ENGLISH));
+            return Format.valueOf(data.optString("format", "COMMAND").toUpperCase(Locale.ENGLISH));
         }
-
-        return format;
     }
 
     public synchronized static PrintProcessor getPrintProcessor(Format format) {
@@ -190,13 +181,24 @@ public class PrintingUtilities {
      * @param params  Params of call from web API
      */
     public static void processPrintRequest(Session session, String UID, JSONObject params) throws JSONException {
-        Format format = getPrintFormat(params.getJSONArray("data"));
+        JSONArray printData = params.getJSONArray("data");
+        convertVersion(printData);
+
+        // grab first data object to determine type for entire set
+        JSONObject firstData = printData.optJSONObject(0);
+        Type type = getPrintType(firstData);
+        Format format = getPrintFormat(type, firstData);
+
         PrintProcessor processor = PrintingUtilities.getPrintProcessor(format);
         log.debug("Using {} to print", processor.getClass().getName());
 
         try {
             PrintOutput output = new PrintOutput(params.optJSONObject("printer"));
             PrintOptions options = new PrintOptions(params.optJSONObject("options"), output, format);
+
+            if(type != Type.RAW && !output.isSetService()) {
+                throw new Exception(String.format("%s cannot print to a raw %s", type, output.isSetFile() ? "file" : "host"));
+            }
 
             processor.parseData(params.getJSONArray("data"), options);
             processor.print(output, options);

